test/connectors/ServiceInfoPartialConnectorSpec.scala

```diff
@@ -32,45 +32,78 @@ import scala.concurrent.{ExecutionContext, Future}
 class ServiceInfoPartialConnectorSpec extends ControllerBaseSpec {
   val header: HeaderCarrierForPartialsConverter = injector.instanceOf[HeaderCarrierForPartialsConverter]
   val btanl: BtaNavigationLinks = injector.instanceOf[BtaNavigationLinks]
-  val validHtml: Html = Html("<nav>BTA lINK</nav>")
-
-  private trait Test {
-    val result :Future[HtmlPartial] = Future.successful(Success(None,validHtml))
-    val httpClient: HttpClient = mock[HttpClient]
-    lazy val connector: ServiceInfoPartialConnector = {
-
-      (httpClient.GET[HtmlPartial](_: String, _: Seq[(String, String)], _: Seq[(String, String)])
-                                  (_: HttpReads[HtmlPartial],_: HeaderCarrier,_: ExecutionContext))
-        .stubs(*,*,*,*,*,*)
-        .returns(result)
-      new ServiceInfoPartialConnector(httpClient, header, btanl)(messagesApi, mockAppConfig)
-    }
-
-  }
+  lazy val validHtml: Html = Html("<nav>BTA lINK</nav>")
+  lazy val result :Future[HtmlPartial] = Future.successful(Success(None,validHtml))
+  val httpClient: HttpClient = mock[HttpClient]
 
   "ServiceInfoPartialConnector" should {
-    "generate the correct url" in new Test {
+
+    "generate the correct url" in {
+
+      lazy val connector: ServiceInfoPartialConnector = {
+
+        (httpClient.GET[HtmlPartial](_: String, _: Seq[(String, String)], _: Seq[(String, String)])
```

Comment:
Could you put this mocking and connector creation in a little function in this file? Just conscious it's being repeated in every test here with the only change being the value that is returned.